### PR TITLE
[WIP] Idm sdk base

### PR DESCRIPTION
**Pull request by wei-lee**
### Motivation

https://issues.jboss.org/browse/AEROGEAR-2086

### Description

Create some of the base classes/interfaces for the iOS IDM SDK based on the android one.

### Progress

- [x] Add classes
- [x] Fix code style

Please see commit ca691d9. The other one is auto generated by swiftlint.

ping @aidenkeating @rachael-oregan @ziccardi could you take a look at this PR if you are planning on the ios IDM SDK. We will chat about it during the standup today.

### CODE REVIEW

File: modules/auth/AgsAuth.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        authServiceConfig = authConfig
+        credentialManager = CredentialsManager()
+        authenticator = OIDCAuthenticator()
+    }
+
+    public func login(presentingViewController: UIViewController, onCompleted: @escaping (User?, Error?) -> Void) {
+
+    }
+
+    public func resumeAuth(url: URL) -> Bool {
```
Comment (aidenkeating) [inline modules/auth/AgsAuth.swift]: What is this method for?

Comment (wei-lee) [reply modules/auth/AgsAuth.swift]: similar to the authentication flow on android, when user logged in via the browser, the app will be resumed. This method should be used when the app is resumed to completed the login process.

Comment (aidenkeating) [reply modules/auth/AgsAuth.swift]: Ah yep, makes sense now, thanks

**Comment by wtrocki** [comment]
@wei-lee You will need to install swift lint on your machine. See ./scripts/develop.sh for details.

File: modules/auth/authenticator/Authenticator.swift [hunk 2]
```diff
@@ -0,0 +1,10 @@
+//
+//  Authenticator.swift
+//  AGSAuth
+
+import Foundation
+
+public protocol Authenticator {
```
Comment (wtrocki) [inline modules/auth/authenticator/Authenticator.swift]: Good style is to have protocols ending ..Able. Not big deal. Just mentioning

Comment (wei-lee) [reply modules/auth/authenticator/Authenticator.swift]: I don't know if that's possible for all the protocols?

Comment (wtrocki) [reply modules/auth/authenticator/Authenticator.swift]: Yeah. It will be something like `Authenticable` and that sounds strange! Let's ignore this.

**Pull request merged**